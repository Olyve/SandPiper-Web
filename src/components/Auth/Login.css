.login {
  display: flex;
  flex-direction: column;
  margin: 10px;
  background-color: #ccc;
  border-radius: 10px;
  width: 300px;
  height: 175px;
}

form {
  flex: 4;
  display: flex;
  flex-direction: column;
}

form div {
  flex: 1;
  height: 20px;
}

h2 {
  flex: 1;
  color: #222;
}