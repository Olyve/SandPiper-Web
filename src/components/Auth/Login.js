import React, { Component } from 'react';
import './Login.css';

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };

    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  onChange(event) {
    let target = event.target;
    let value = target.value;
    let name = target.name;

    this.setState({
      [name]: value
    });
  }

  handleSubmit(event) {
    this.props.handleLogin({
      email: this.state.email,
      password: this.state.password
    });
    this.setState({
      email: '',
      password: ''
    });

    event.preventDefault();
  }

  render() {
    return (
      <div className='login'>
        <h2>Login</h2>
        <form>
          <div>
            <label>Email: </label>
            <input type='text' name='email' value={this.state.email} onChange={this.onChange} />
          </div>
          <div>
            <label>Password: </label>
            <input type='password' name='password' value={this.state.password} onChange={this.onChange} />
          </div>
          <div>
            <button className='submit' onClick={this.handleSubmit}>Submit</button>
          </div>
        </form>
      </div>
    );
  }
}

export default Login;