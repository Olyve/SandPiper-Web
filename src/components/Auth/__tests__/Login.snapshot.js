import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../Login';

describe('Login', () => {
  it('should render correctly', () => {
    const rendered = renderer.create(
      <Login />
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});