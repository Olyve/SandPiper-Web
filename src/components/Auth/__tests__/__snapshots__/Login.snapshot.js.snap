// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<div
  className="login"
>
  <h2>
    Login
  </h2>
  <form>
    <label>
      Email:
      <input
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <br />
    <label>
      Password:
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </label>
    <br />
    <button
      className="submit"
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
</div>
`;
